HSP2tools/readWDM.py

```diff
@@ -149,137 +171,199 @@ def readWDM(wdmfile, hdffile, jupyterlab=True):
+        converted_dates = []
+        for x in dates:
+            year, month, day, hour, minute, second = bits_to_date(x)
+            date = date_epoch
+            date += (year - 1970) * dt_year
+            date += (month - 1) * dt_month
+            date += (day - 1) * dt_day
+            date += hour * dt_hour
+            date += minute * dt_minute
+            date += second * dt_second
+            converted_dates.append(date)
+        return converted_dates
+
+    #@njit
+    def _process_groups(self, iarray, farray, records, offsets, tgroup):
+        date_array = [0] #need initialize with a type for numba
+        value_array = [0.0]
+
+        for i in range(0,len(records)):
+            record = records[i]
+            offset = offsets[i]
+            index = record * 512 + offset
+            pscfwr = iarray[record * 512 + 3] #should be 0 for last record in timeseries 
+            current_date = WDMReader._splitdate(iarray[index].item())
+            group_enddate = WDMReader.increment_date(current_date, tgroup, 1)
+            offset +=1
+            index +=1
+
+            while current_date < group_enddate:
+                nval, ltstep, ltcode, comp, qual = WDMReader._splitcontrol(iarray[index])  
+                #compressed - only has single value which applies to full range
+                if comp == 1:
+                    for i in range(0, nval, 1):
+                        current_date = WDMReader.increment_date(current_date, ltcode, ltstep.item()) 
+                        date_array.append(current_date)
+                        value_array.append(farray[index + 1])
+                    index += 2
+                    offset +=2
+                else:
+                    for i in range(0, nval, 1):
+                        current_date = WDMReader.increment_date(current_date, ltcode, ltstep.item()) 
+                        date_array.append(current_date)
+                        value_array.append(farray[index + 1 + i])
+                    index += 1 + nval
+                    offset +=1 + nval
+                
+                if offset >= 512:
+                    offset = 4
+                    index = (pscfwr - 1) * 512 + offset
+                    record = pscfwr
+                    pscfwr = iarray[(record - 1) * 512 + 3] #should be 0 for last record in timeseries
+
+        date_array = date_array[1:]
+        value_array = value_array[1:]
+
+        return date_array, value_array
+
+reader = WDMReader("../WDM/rpo772.wdm")
+#reader.readWDM("../WDM/rpo772.wdm", "rpo772.h5", True)
+def func():
+    reader.readWDM()
+    reader.WriteHD5("rpo772.h5", True)
+#     reader = WDMReader("../WDM/rpo772.wdm")
+#     reader.readWDM("../WDM/rpo772.wdm", "rpo772.h5", True)
+#     #reader.readWDM("../WDM/test.wdm", "test.h5")
+#     #reader.readWDM("../WDM/2_RPO_SWMM48LINKS2017_CBOD.wdm", "2_RPO_SWMM48LINKS2017_CBOD.h5")
+
+execution_time = timeit.timeit(func, number=1)
+
+print(execution_time)
```

Comment:
I gather this is all for testing which is fine. We'll just need to make sure we remove this before we merge this back into the development branch.